src/index.ts

```diff
@@ -180,13 +199,18 @@ namespace WebComponentsManager {
     }
 
     export function whenDefined<T>(obj: object, key: string | symbol): Promise<T> {
+
       if (!obj.hasOwnProperty(key)) {
         let setter;
         let getter: any = new Promise<T>((resolve): void => {
-          setter = (val: T) => (getter = val, setter = () => null, resolve(val));
+          setter = (val: T) => {
+            getter = val;
+            setter = () => null;
+            resolve(val);
+          };
         });
 
-        Object.defineProperty(obj, key, { get: () => getter, set: setter });
+        Object.defineProperty(obj, key, { get: () => getter, set: setter, configurable: true });
```

Comment:
Can the `configurable: true` be removed?

Reply:
If not, I can look into this later on.

Reply:
yes. I have tested and works just as well.